src/mongo_odm/model.hpp

```diff
@@ -84,6 +160,133 @@ class model {
     }
 
     /**
+     * Finds the documents in this collection which match the provided filter.
+     *
+     * @param filter
+     *   Document view representing a document that should match the query.
+     * @param options
+     *   Optional arguments, see mongocxx::options::find
+     *
+     * @return Cursor with deserialized objects from the collection.
+     * @throws
+     *   If the find failed, the returned cursor will throw mongocxx::exception::query when it
+     *   is iterated.
+     *
+     * @see https://docs.mongodb.com/manual/tutorial/query-documents/
+     */
+    static deserializing_cursor<T> find(
+        bsoncxx::document::view_or_value filter,
+        const mongocxx::options::find& options = mongocxx::options::find()) {
+        return _coll.find(std::move(filter), options);
+    }
+
+    /**
+     * Finds a single document in this collection that matches the provided filter.
+     *
+     * @param filter
+     *   Document view representing a document that should match the query.
+     * @param options
+     *   Optional arguments, see mongocxx::options::find
+     *
+     * @return An optional object that matched the filter.
+     * @throws mongocxx::exception::query if the operation fails.
+     *
+     * @see https://docs.mongodb.com/manual/tutorial/query-documents/
+     */
+    static mongocxx::stdx::optional<T> find_one(
+        bsoncxx::document::view_or_value filter,
+        const mongocxx::options::find& options = mongocxx::options::find()) {
+        return _coll.find_one(std::move(filter), options);
+    }
+
+    /**
+     *  Inserts multiple object of the model into the collection.
+     *
+     *  @warning This method uses the bulk insert command to execute the insertion as opposed to
+     *  the legacy OP_INSERT wire protocol message. As a result, using this method to insert
+     *  many documents on MongoDB < 2.6 will be slow.
+     *
+     *  @tparam containter_type
+     *    The container type. Must contain an iterator that yields objects of this model.
+     *
+     *  @param container
+     *    Container of model objects to insert.
+     *  @param options
+     *    Optional arguments, see mongocxx::options::insert.
+     *
+     *  @return The result of attempting to performing the insert.
```

Comment:
add `@see` links to [the insert docs](https://docs.mongodb.com/manual/tutorial/insert-documents/)